src/transport/transport.h

```diff
@@ -92,15 +92,22 @@ class ConvectionTransport : public TransportBase {
         FieldSet output_fields;
     };
 
+
+    typedef ConcentrationTransportBase FactoryBaseType;
+
+    static const Input::Type::Record & get_input_type();
+
     /**
      * Constructor.
      */
-        ConvectionTransport(Mesh &init_mesh, const Input::Record in_rec);
+    ConvectionTransport(Mesh &init_mesh, const Input::Record in_rec);
 	/**
 	 * TODO: destructor
 	 */
 	virtual ~ConvectionTransport();
 
+	void initialize() override;
```

Comment:
Is it really necessary to introduce other initialization method then constructor and zero_time_step?